src/main/drivers/timer_common.c

```diff
@@ -38,37 +38,26 @@ timerIOConfig_t *timerIoConfigByTag(ioTag_t ioTag)
     UNUSED(ioTag);
     return NULL;
 }
-#endif
 
 static uint8_t timerIndexByTag(ioTag_t ioTag)
 {
-#ifdef USE_TIMER_MGMT
     for (unsigned i = 0; i < MAX_TIMER_PINMAP_COUNT; i++) {
         if (timerIOConfig(i)->ioTag == ioTag) {
             return timerIOConfig(i)->index;
         }
     }
-#else
-    UNUSED(ioTag);
-#endif
     return 0;
 }
 
 const timerHardware_t *timerGetByTagAndIndex(ioTag_t ioTag, unsigned timerIndex)
 {
-    if (!ioTag) {
+    if (!ioTag || !timerIndex) {
         return NULL;
     }
 
 #if TIMER_CHANNEL_COUNT > 0
-    uint8_t index = 1;
-    for (unsigned i = 0; i < TIMER_CHANNEL_COUNT; i++) {
-        if (TIMER_HARDWARE[i].tag == ioTag) {
-            if (index == timerIndex || timerIndex == 0) {
-                return &TIMER_HARDWARE[i];
-            }
-            index++;
-        }
+    if (TIMER_HARDWARE[timerIndex - 1].tag == ioTag) {
```

Comment:
`timerIndex` is the index relative to the group of timers for the given `ioTag`, so it cannot be used as a direct index into the list of timers for all ioTags (`TIMER_HARDWARE`).